app/views/directives/osc-persistent-volume-claim.html

```diff
@@ -2,37 +2,43 @@
   <fieldset ng-disabled="claimDisabled">
     <div ng-show="storageClasses | hashSize" class="form-group">
       <!--storage class-->
-      <label class="required">Storage Classes</label>
+      <label>Storage Class</label>
+      <div>
+      <ui-select
+        ng-if="storageClasses"
+        ng-model="claim.storageClass"
+        theme="bootstrap"
+        search-enabled="true"
+        title="Select a storage class"
+        class="select-role"
+        flex>
+        <ui-select-match placeholder="Select a storage class">
+          <span>
+            {{$select.selected.metadata.name}}
+          </span>
+        </ui-select-match>
+        <ui-select-choices
+          repeat="sclass as sclass in storageClasses | toArray | filter : { metadata: { name: $select.search } } | orderBy: 'metadata.name'">
+          <div>{{ sclass.metadata.name }}
+            <span ng-if="sclass | annotation : 'description'" class="text-muted">
+              <small>&nbsp;-
+                <span ng-if="sclass.parameters.type">Type: {{sclass.parameters.type}} | </span>
+                <span ng-if="sclass.parameters.zone">Zone: {{sclass.parameters.zone}} | </span>
+                {{sclass | annotation : 'description'}}
+              </small>
+            </span>
+          </div>
+        </ui-select-choices>
+      </ui-select>
+      </div>
       <div id="claim-storage-class-help" class="help-block mar-bottom-lg">
+        <span ng-if="defaultStorageClass">"{{defaultStorageClass}}" is currently set as the default storage class.<br/></span>
```

Comment:
I'd remove the `<br>` and move this after the description to be consistent with help text we use on other forms.

```html
<div id="claim-storage-class-help" class="help-block mar-bottom-lg">
  Storage classes are set by the administrator to define types of storage the users can select.
  <span ng-if="defaultStorageClass">If not set, default storage class <var>{{defaultStorageClass}}</var> will be used.</span>
</div>
```

I'm curious why `mar-bottom-lg` is needed. I thought the `help-block` and `learn-more-block` styles had appropriate margins. It's better not to set it if we can avoid it so that spacing is consistent across pages.